tests/integrated/test-laplace3d-gridfile/test-laplace3d.cxx

```diff
@@ -0,0 +1,137 @@
+    int x = it.ind;
+    int y = mesh->ystart - 1;
+    if (x == mesh->xstart) {
+      for (int z = mesh->zstart; z <= mesh->zend; z++) {
+	f(x-1, y, z) = 0.5*(f(x-1, y - 1, z) + f(x-1, y, z));
+      }
+    }
+    for (int z = mesh->zstart; z <= mesh->zend; z++) {
+      f(x, y, z) = 0.5*(f(x, y, z) + f(x, y + 1, z));
+    }
+    if (x == mesh->xend) {
+      for (int z = mesh->zstart; z <= mesh->zend; z++) {
+	f(x+1, y, z) = 0.5*(f(x+1, y - 1, z) + f(x+1, y, z));
+      }
+    }
+  }
+  for (auto it = mesh->iterateBndryUpperY(); !it.isDone(); it.next()) {
+    int x = it.ind;
+    int y = mesh->yend + 1;
+    if (x == mesh->xstart) {
+      for (int z = mesh->zstart; z <= mesh->zend; z++) {
+	f(x-1, y, z) = 0.5*(f(x-1, y - 1, z) + f(x-1, y, z));
+      }
+    }
+    for (int z = mesh->zstart; z <= mesh->zend; z++) {
+      f(x, y, z) = 0.5*(f(x, y - 1, z) + f(x, y, z));
+    }
+    if (x == mesh->xend) {
+      for (int z = mesh->zstart; z <= mesh->zend; z++) {
+	f(x+1, y, z) = 0.5*(f(x+1, y - 1, z) + f(x+1, y, z));
+      }
+    }
+  }
+  if (mesh->firstX()) {
+    int x = mesh->xstart - 1;
+    for (int y = mesh->ystart; y <= mesh->yend; y++) {
+      for (int z = mesh->zstart; z <= mesh->zend; z++) {
+        f(x, y, z) = 0.5*(f(x, y, z) + f(x, y + 1, z));
+      }
+    }
+  }
+  if (mesh->lastX()) {
+    int x = mesh->xend + 1;
+    for (int y = mesh->ystart; y <= mesh->yend; y++) {
+      for (int z = mesh->zstart; z <= mesh->zend; z++) {
+        f(x, y, z) = 0.5*(f(x, y - 1, z) + f(x, y, z));
+      }
+    }
+  }
+
+  ///////////////////////////////////////////////////////////////////////////////////////
+  // Set up Laplace solver
+  ///////////////////////////////////////////////////////////////////////////////////////
+  auto laplace_solver = Laplacian::create();
+  Field3D A, C1, C2, D;
+  initial_profile("A", A);
+  initial_profile("C1", C1);
+  initial_profile("C2", C2);
+  mesh->communicate(C2);
+  C2.setBoundary("C2");
+  C2.applyParallelBoundary();
+  initial_profile("D", D);
+  laplace_solver->setCoefA(A);
+  laplace_solver->setCoefC1(C1);
+  laplace_solver->setCoefC2(C2);
+  laplace_solver->setCoefD(D);
+
+  ///////////////////////////////////////////////////////////////////////////////////////
+  // Solve
+  ///////////////////////////////////////////////////////////////////////////////////////
+  f = laplace_solver->solve(rhs, f);
+
+  ///////////////////////////////////////////////////////////////////////////////////////
+  // Calculate error
+  ///////////////////////////////////////////////////////////////////////////////////////
+  Field3D rhs_check = D*Laplace_perp(f) + Grad_perp(C2)*Grad_perp(f)/C1 + A*f;
+  //auto& coords = *mesh->getCoordinates();
+  //auto& g11 = coords.g11;
+  //auto& g33 = coords.g33;
+  //Field3D rhs_check = D*Delp2(f, CELL_DEFAULT, false)
+  //                    + (g11*DDX(C2)*DDX(f) + g33*DDZ(C2)*DDZ(f))/C1
+  //                    + A*f;
```

Comment:
Commented out code

Reply:
This is something which I sometimes use for testing purposes. I'll delete it once I have all integrated tests passing.